Marlin/src/HAL/STM32/MarlinSerial.cpp

```diff
@@ -87,11 +87,12 @@
 #endif
 
 void MarlinSerial::begin(unsigned long baud, uint8_t config) {
-  HardwareSerial::begin(baud, config);
+  HAL_HardwareSerial::begin(baud, config); // IRON, MODIFIED TO "HAL_HardwareSerial"
   // Replace the IRQ callback with the one we have defined
   TERN_(EMERGENCY_PARSER, _serial.rx_callback = _rx_callback);
 }
 
+// IRON, IT SEEMS THIS CODE IS NEVER USED (STM32)??????????
```

Comment:
There are a few references to `_rx_complete_irq` in the code:
```cpp
  if (baud == WIFI_BAUDRATE)
    uart_attach_rx_callback(&_serial, _rx_complete_irq);
  else
    USART1->CR1 |= USART_CR1_RE;  // Preserve word length, etc. Use 'or' to preserve USART_CR1_M_8N1
```
```cpp
#define DECLARE_SERIAL_PORT(ser_num) \
  void _rx_complete_irq_ ## ser_num (serial_t * obj); \
  MSerialT MSerial ## ser_num (true, USART ## ser_num, &_rx_complete_irq_ ## ser_num); \
  void _rx_complete_irq_ ## ser_num (serial_t * obj) { MSerial ## ser_num ._rx_complete_irq(obj); }
```

Reply:
> There are a few references to _rx_complete_irq in the code:

Yes, I saw, but I never enable the interrupt callbacks, so it never gets called when using DMA serial reading. This is what I want, so I don't need to change the code there.